validate/validators/bam.py

```diff
@@ -14,3 +15,13 @@ def validate_bam_file(path):
         raise ValueError("pysam bam check failed. No reads in " + str(path))
 
     return True
+
+def check_bam_index(path):
+    '''Checks if index file is present and can be opened'''
+    try:
+        pysam.AlignmentFile(str(path)).check_index()
+    except ValueError:
+        raise AttributeError(f'''pysam bam index check failed. Index file for {str(path)} could not
```

Comment:
`AttributeError` is not the correct exception for this. https://docs.python.org/3/library/exceptions.html#AttributeError

> Raised when an attribute reference (see [Attribute references](https://docs.python.org/3/reference/expressions.html#attribute-references)) or assignment fails.

`FileNotFoundError` may be more appropriate. https://docs.python.org/3/library/exceptions.html#FileNotFoundError

These ones might work too:

- `ValueError` (commonly used when there isn't a more specific error) https://docs.python.org/3/library/exceptions.html#ValueError
- `EnvironmentError` https://docs.python.org/3/library/exceptions.html#EnvironmentError
- `IOError` https://docs.python.org/3/library/exceptions.html#IOError